### PR TITLE
fix: Chat input bugs and refactor it

### LINKED ISSUE #588: [BUG]: Fix chat input bugs and refactor it. Fix command list and member list.
### Description:
Fixing many chat input issues and refactoring the component.

### Expected behavior:
Many things don't work properly, especially the command list. The code is hard to understand because it's not modular.

### Actual behavior:
Things should work as expected.

### PR DESCRIPTION
**Pull request by Spiral-Memory**
# Brief Title: Fix Chat Input Bugs and Refactor This Component

## Acceptance Criteria Fulfillment

- [x] Fixed MemberList and CommandList stylings.
- [x] Added auto-scroll functionality to MentionList and CommandList menus.
- [x] Added keyboard control in CommandList menu.
- [x] Implemented word break in the input component when text exceeds the limit.
- [x] Added scroll functionality in the input.
- [x] Fixed Ctrl+B and Ctrl+I usage and toggling and tried reusing the existing functionality 
- [x] Refactored code into separate functions and components based on their concerns.

Fixes #588

## Video/Screenshots

Before: (Issues mentioned descriptively in issue page)

https://github.com/RocketChat/EmbeddedChat/assets/78961432/580e17e0-2efc-479a-8683-d595bb7cdb8b

After: 

https://github.com/RocketChat/EmbeddedChat/assets/78961432/9a3a89ed-db08-4f76-8891-9f427bdfc183

### CODE REVIEW

File: packages/react/src/hooks/useFormatSelection.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+const useFormatSelection = (messageRef) => {
+  const formatSelection = (pattern) => {
```
Comment (sidmohanty11) [inline packages/react/src/hooks/useFormatSelection.js:2]: can we separate out this function to a file and add some basic unit tests to it if possible?

Comment (Spiral-Memory) [reply packages/react/src/hooks/useFormatSelection.js:2]: It's already in a different file Sid, regarding unit test, I'll add it later.

File: packages/react/src/views/Mentions/MembersList.js [hunk 2]
```diff
@@ -1,47 +1,111 @@
...
+  setShowMembersList,
+}) {
+  const itemRefs = useRef([]);
   const styles = useMemberListStyles();
   const { colors } = useCustomTheme();
 
   const handleMemberClick = useCallback(
     (selectedItem) => {
-      onMemberClick(selectedItem);
+      let insertionText;
```
Comment (sidmohanty11) [inline packages/react/src/views/Mentions/MembersList.js:22]: lets extract it to a function

File: packages/react/src/hooks/useFormatSelection.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+      const wrappedText = startPattern + selectedText + endPattern;
+      input.value = initText + wrappedText + finalText;
+      input.selectionStart = selectionStart + startPattern.length;
+      input.selectionEnd = input.selectionStart + selectedText.length;
+    }
+
+    input.focus();
+  };
+
+  return { messageRef, formatSelection };
```
[the remark this answers is not in the document]
Comment (Spiral-Memory) [reply packages/react/src/hooks/useFormatSelection.js:37]: Didn't get you ? Can you explain a little ?

**Review by sidmohanty11 — APPROVED** [review]

**Pull request merged**